browser/resources/settings/shortcuts_page/utils/commandsApi.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import {
+  createInterfaceApi,
+  eventsFor,
+  endpointsFor,
+  state,
+} from '$web-common/api'
+
+import * as Mojom from 'gen/brave/components/commands/common/commands.mojom.m'
+
+type Actions = Pick<
+  Mojom.CommandsServiceInterface,
+  | 'resetAccelerators'
+  | 'resetAcceleratorsForCommand'
+  | 'assignAcceleratorToCommand'
+  | 'unassignAcceleratorFromCommand'
+>
+
+export function createCommandsApi(
+  controller: Mojom.CommandsServiceRemote,
+) {
+
+  // I feel a bit hazy on when I should use an action / endpoint / event. This is my guess:
+  // 1. Actions are for methods which have no return value
+  // 2. Endpoints are for getting a response (and potentially caching the value & sharing that value)
```

Comment:
That's largely right. You _can_ have an endpoint that doesn't do any fetch - a `state` endpoint and that's where the word endpoint fails, but that's only to provide a shape and intial version of data that's provided only by events.

I think this should be clearer in the docs:
query: immediately fetches when `useXYZ()` is called
mutation: only fetched when `useXYZ().mutate()` is called